ci/do_ci.sh

```diff
@@ -367,6 +367,7 @@ elif [[ "$CI_TARGET" == "check_format" ]]; then
   echo "check_format_test..."
   ./tools/code_format/check_format_test_helper.sh --log=WARN
   echo "check_format..."
+  ./tools/code_format/check_shellcheck_format.sh
```

Comment:
Can it do some auto fix?

Reply:
it has a `-f diff` flag, but i dont think you would want it to autofix.

Shellcheck can often get false positives and the suggestions will often not work the same as the code ime.

i can add the diff flag at least - i had originally not intended to output the shellcheck output, just the result - but some things were failing differently on azure due to different shellcheck version, so i added it back.

ill update now with diff, and some small cleanup...

Reply:
hmm, it turns out the version of shellcheck in bionic doesnt have diff:

https://dev.azure.com/cncf/envoy/_build/results?buildId=49942&view=logs&j=9fd74226-df5b-5f29-39da-76fc341cc763&t=068b98b1-acf0-52d0-fc6e-f6d1a5b1d5ef&l=113

https://dev.azure.com/cncf/envoy/_build/results?buildId=49942&view=logs&j=9fd74226-df5b-5f29-39da-76fc341cc763&t=068b98b1-acf0-52d0-fc6e-f6d1a5b1d5ef&l=132

https://github.com/koalaman/shellcheck/issues/1220#issuecomment-615030110

Reply:
Ah ok, then lets leave this as it is, can you add a TODO, so we can have that when we update to newer version of shellcheck?

Reply:
yep